web-ng/src/app/components/map/map.component.ts

```diff
@@ -124,21 +120,26 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         );
       case EditMode.AddPolygon:
         // TODO: Implement adding polygon.
+        if (!selectedLayerId) {
+          return Promise.resolve();
+        }
         return Promise.resolve();
       case EditMode.None:
       default:
+        this.navigationService.setFeatureId(null);
         return Promise.resolve();
     }
   }
 
   private onProjectAndFeaturesUpdate(
     project: Project,
     features: List<Feature>,
-    selectedFeatureId: string | null
+    selectedFeatureId: string
   ): void {
     this.removeMarkersAndGeoJsonsOnMap(features);
-    this.addMarkersAndGeoJsonsToMap(project, features, selectedFeatureId);
+    this.addMarkersAndGeoJsonsToMap(project, features);
```

Comment:
Probably out of scope of this PR, but at some point might be worth renaming this to "addFeaturesToMap()`.

Reply:
We can do it in this one, the scope of this function is rather small. Extracted.